### PR TITLE
Delete evaluations

### PR DESCRIPTION
**Pull request by Abriscioli**
@andresmoschini 

Hi Andrés!

I finished the methods to delete evaluations. I also created a new role, EvaluationManager, who is going to be able to delete those documents. Could you review?

### CODE REVIEW

File: source/CommonJobs/CommonJobs.Application.EvalForm/Commands/DeleteEvaluationCommand.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            RavenSession.Advanced.DocumentStore.DatabaseCommands.DeleteByIndex<EvaluationToDelete_Search>(indexQuery).WaitForCompletion();
+            return true;
+        }
+
+        /// <summary>
+        /// Escapes symbols like * or " to avoid deleting more data than intended
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static string Escape(string value)
```
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.Application.EvalForm/Commands/DeleteEvaluationCommand.cs]: In the future, consider to move it to a heleper class to allow reusing. By the moment it is fine here.

File: source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CommonJobs.Application.EvalForm.EmployeeSearching
+{
+    class EvaluationToDelete_Search : AbstractMultiMapIndexCreationTask<EvaluationToDelete_Search.Projection>
+    {
+        public class Projection
```
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs]: Is it necessary to define a projection? maybe a simpler index is enough

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/dashboard.js [hunk 3]
```diff
@@ -169,7 +169,8 @@
         this.items(_.map(data.Evaluations, function (e) {
             return new Evaluation(e);
         }));
-    }
+        this.period = data.Period;
```
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/dashboard.js]: If you will not use it, I thik that it is better to revert the change.

If you will no revert, I suggest you to consider define the field as default in constructor and maybe using an observable.

File: source/CommonJobs/CommonJobs.MVC.UI/Areas/Evaluations/Controllers/EvaluationsApiController.cs [hunk 4]
```diff
@@ -53,7 +53,7 @@ public JsonNetResult GenerateEvalutions(PeriodCreation model)
         [AcceptVerbs(HttpVerbs.Get)]
         public JsonNetResult GetDashboardEvaluationsForEmployeeManagers(string period)
         {
-            PeriodEvaluation periodEvaluation = new PeriodEvaluation();
+            PeriodEvaluation periodEvaluation = new PeriodEvaluation(period);
```
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.MVC.UI/Areas/Evaluations/Controllers/EvaluationsApiController.cs]: If you will not use it, I recommend you to revert these changes.

File: source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs (hunk 2 above)
Comment (Abriscioli) [inline source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs]: I'm trying to avoid the projection, but i think it's clearer to see the result structure.

Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs]: In that case, you should try to create _projection type_ objects as result of map and reduce operations. Anyway it does not seems to be a RavenDB common practice.

Anyway, I was not talking about projection itself, I was talking about the index type (`AbstractMultiMapIndexCreationTask`), there should be a simpler index base class that fits for your needs.

Comment (Abriscioli) [inline source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EvaluationToDelete_Search.cs]: Well, I can't avoid using `AbstractMultiMapIndexCreationTask`, but the projection is not needed for deleting the records, so we can remove it
